src/mavedb/lib/validation/keywords.py

```diff
@@ -1,8 +1,47 @@
+from sqlalchemy.orm import Session
+from typing import Optional
+
 from mavedb.lib.validation.exceptions import ValidationError
 from mavedb.lib.validation.utilities import is_null
 
+from mavedb.models.controlled_keyword import ControlledKeyword
+
+
+# value will not be Optional when we confirm the final controlled keyword list.
+def find_keyword(db: Session, key: str, value: Optional[str]):
+    query = db.query(ControlledKeyword).filter(ControlledKeyword.key == key).filter(ControlledKeyword.value == value)
+    controlled_keyword = query.one_or_none()
+    if controlled_keyword is None:
+        raise ValueError(f'Invalid keyword {key} or {value}')
+    return controlled_keyword
```

Comment:
Perhaps we could move this into it's own `mavedb/lib/keywords.py` file, just like how experiments and score sets have a `search_{resource}` function? Stylistically I think the functions in the validation file shouldn't interact with the database at all.